quizz.md

```diff
@@ -0,0 +1,213 @@
+
+```
+Belle is friends with Chip Potts
+Belle is friends with Cogsworth
+Belle is friends with Lumière
+Belle is friends with Mrs. Potts
+```
+
+Write your code here:
+```ruby
+
+friends.each do |f|
+  puts "Belle is friends with " + f
+end
+
+```
+
+## SQL, Databases, and ActiveRecord (meets Aladdin)
+
+### Question 4
+
+Describe what an ERD is, and why we create them for applications. Also give an
+example what the attributes and relationships might be for the following
+entities (no need to draw an ERD):
+<!-- Maybe clarify whether they're meant to give relationships between all four entities or... --> # I'm having a hard time understanding what you mean by this...
+
+* Genie
+* Lamp
+* Person
+* Pet
+
+Your answer:
+```
+ An entity-relationship (ER) diagram is a specialized graphic that illustrates the relationships between entities in a database. ER diagrams often use symbols to represent three different types of information. Boxes are commonly used to represent entities. Diamonds are normally used to represent relationships and ovals are used to represent attributes.
+
+ A person can have many Lamps and each lamp one Genie?
+ and a person can have one pet?... I'm unsure about how to explain this.
+```
+
+### Question 5
+
+Describe what a schema is, and how we represent a one-to-many relationship in a
+SQL database. If you need an example, you can use: people and wishes
+(one-to-many).
+
+Your answer:
+```
+A schema is a collection of database objects (as far as this hour is concerned—tables) associated with one particular database username. This username is called the schema owner, or the owner of the related group of objects. You may have one or multiple schemas in a database.
+
+```
+
+### Question 6
+
+**Assume:**
+1. Your database two working tables, `genies` and `lamps`.
+2. You have a working connection to the database for ActiveRecord.
+3. You have active record models defined for `Genie` and `Lamp`, and the
+relationships between the two are set up in Active Record.
+<!-- Do we want to specifiy what kind of relationship they have, in case some students aren't familiar with the mythology...? -->
+4. Lamps have one property, `wishes_remaining`, and genies have one property, `name`.
+
+Write code to do the following:
+
+1. Create a lamp with 3 wishes remaining and a genie named 'Genie'
+2. Create a relationship between 'Genie' and the lamp.
+3. Update the lamp so it only has one wish left.
+  * Oh no... Jafar has Aladdin! Thankfully he's wished to become a genie himself!
+4. Create a new Genie named 'Jafar' and put him in a new lamp with 3 wishes left.
+5. Free the good Genie by setting his lamp to `nil`
+
+
+Write your code here:
+```ruby
+# code here
+Lamp.create(wishes_remaining: 3)
+
+Genie.create(name: "Genie")
+
+class Genie < ActiveRecord::Base
+  has_one :lamp
+end
+
+class Lamp < ActiveRecord::Base
+  belongs_to :genie
+end
+
+no_one_lamp = Lamp.create(id: 1, wishes_left: 3, genie_id: 1)
```

Comment:
- We rarely (never?) assign our own ID.  Let the db do that ensure every new item has a unique id.
- recommend using the object to create the relationship.  We want to avoid working with id directly.  Let rails to that.
    ```
    genie = Genie.create(name: "Genie")
    no_one_lamp = Lamp.create(wishes_left: 3, genie: genie)`
    ```
    Or, better yet...
    ```
    genie = Genie.create(name: "Genie")
    genie.lamp.create(wishes_left: 3)
    ```